// @angelgoezg/common/build/models/quality-test.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const mongoose_1 = __importDefault(require("mongoose"));
const { Schema } = mongoose_1.default;
const QualityTestSchema = new Schema({
    analyst: {
        type: Schema.Types.ObjectId,
        ref: 'Analista'
    },
    contract: {
        type: Schema.Types.ObjectId,
        ref: 'Contrato'
    },
    year: {
        type: String,
        required: true,
        trim: true
    },
    zoneManager: {
        type: Schema.Types.ObjectId,
        ref: 'CoordiandorZona',
        required: true
    },
    month: {
        type: String,
        required: true,
        trim: true
    },
    weekNumber: {
        type: Number,
        required: true
    },
    verificationDate: {
        type: Date,
        required: true
    },
    grade: {
        type: String,
        required: true,
        trim: true
    },
    auxiliary: {
        type: Schema.Types.ObjectId,
        ref: 'Analista',
        required: true
    },
    registeredBy: {
        type: Schema.Types.ObjectId,
        required: false,
        ref: 'Analista'
    },
    program: {
        type: Schema.Types.ObjectId,
        required: false,
        ref: 'ProgramaActivos'
    },
    sumErrors: {
        type: Number
    },
    caseNumber: {
        type: String,
        required: true,
        trim: true
    },
    idTransaction: {
        type: String,
        required: true,
        trim: true
    },
    previousScale: {
        type: String,
        trim: true
    },
    registers: [
        {
            tipoVerificacion: {
                type: String,
                required: true,
                trim: true
            },
            tipologiaTransaccion1: {
                type: String,
                required: true,
                trim: true
            },
            tipologiaTransaccion2: {
                type: String,
                required: true,
                trim: true
            },
            tipologiaTransaccion3: {
                type: String,
                required: true,
                trim: true
            },
            tipologiaTransaccion4: {
                type: String,
                required: true,
                trim: true
            },
            tipologiaTransaccion5: {
                type: String,
                required: true,
                trim: true
            },
            tipologiaTransaccion6: {
                type: String,
                required: true,
                trim: true
            },
            tipologiaTransaccion7: {
                type: String,
                required: true,
                trim: true
            },
            necesidadFormativa1: {
                type: String,
                required: false,
                trim: true
            },
            necesidadFormativa2: {
                type: String,
                required: false,
                trim: true
            },
            variable: {
                type: Schema.Types.ObjectId,
                ref: 'Variable',
            },
            fechaCorte: {
                type: Date,
                required: false
            },
            sumaCausas: {
                type: Number,
                required: true
            },
            particularidad1: {
                type: String,
                required: true,
                trim: true
            },
            particularidad2: {
                type: String,
                required: true,
                trim: true
            },
            tipoAuditoria: {
                type: String,
                required: true,
                trim: true
            },
            grupo: {
                type: String,
                required: true,
                trim: true
            },
            motivoOrigen: {
                type: String,
                required: true,
                trim: true
            },
            proceso: {
                type: String,
                required: true
            },
            sumECUF: {
                type: Number,
                required: true
            },
            sumECN: {
                type: Number,
                required: true
            },
            sumENC: {
                type: Number,
                required: true
            },
            sumEP: {
                type: Number,
                required: true
            },
            comentario: {
                type: String,
                trim: true,
                required: true
            },
            valorVariable: {
                type: String,
                trim: true
            },
            resultadoVariable: {
                type: Number,
                trim: true
            },
            causa1: {
                type: Schema.Types.ObjectId,
                ref: 'Cause'
            },
            causa2: {
                type: Schema.Types.ObjectId,
                ref: 'Cause'
            },
            causa3: {
                type: Schema.Types.ObjectId,
                ref: 'Cause'
            },
            satisfaccion: {
                type: String,
                trim: true
            },
            FCR: {
                type: String,
                trim: true
            }
        }
    ]
}, { collection: 'checks_vt', timestamps: true });
const Check = mongoose_1.default.model('CheckVT', QualityTestSchema);
exports.default = Check;
